packages/caravan-fees/src/types.ts

```diff
@@ -0,0 +1,312 @@
+import { Network } from "@caravan/bitcoin";
+import {
+  BtcTxInputTemplate,
+  BtcTxOutputTemplate,
+} from "./btcTransactionComponents";
+
+/**
+ * Represents an Unspent Transaction Output (UTXO).
+ * UTXOs are used as inputs for new transactions and are crucial for fee bumping operations.
+ */
+export interface UTXO {
+  /**
+   * The transaction ID of the UTXO.
+   * This is the unique identifier of the transaction that created this output.
+   */
+  txid: string;
+
+  /**
+   * The output index of the UTXO in its parent transaction.
+   * This, combined with the txid, uniquely identifies the UTXO.
+   */
+  vout: number;
+
+  /**
+   * The value of the UTXO in satoshis.
+   * This represents the amount of bitcoin contained in this output.
+   */
+  value: number;
+
+  /**
+   * The locking script of the UTXO.
+   * This script defines the conditions that must be met to spend this output.
+   */
+  script: Buffer;
+
+  /**
+   * Optional field for any additional data required for the input.
+   * This can include things like redeem scripts for P2SH outputs or witness scripts for P2WSH.
+   */
+  additionalData?: any;
```

Comment:
This should be more definitively defined. you can just add `redeemScript` and `witnessScript` as optional properties. If there are other additional items that are needed let's define them otherwise keep it clear what is expected for a utxo.

Reply:
cleaned [now](https://github.com/caravan-bitcoin/caravan/pull/114/commits/09cceac907eb046f8975d6bb43b047c65afef421#diff-593f8d602f801c56261ed7a8dc941947102f6a7b26c483c5ee8cf9c293fffb33R35)